File: leetcode/rust/two-sum.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        use std::collections::HashMap;

        let mut map: HashMap<i32, i32> = HashMap::new();

        for (idx, val) in nums.iter().enumerate() {
            match map.get(val) {
                Some(&i) => return vec![i, idx as i32],
                None => map.insert(target - val, idx as i32),
            };
        }

        panic!("No answer!");
    }
}

File: leetcode/go/two-sum.go
package main

func TwoSum(nums []int, target int) []int {
	hashMap := make(map[int]int)

	for i, num := range nums {
			if v, ok := hashMap[target - num]; ok {
					return []int{v, i}
			}

			hashMap[num] = i
	}

	return nil
}